### PR TITLE
Cumulus 678 refactors move granules task

### PR DESCRIPTION
**Pull request by flamingbear**
**Summary:**  refactor to have move-granules task use same functions as api/granules/move

Addresses [CUMULUS-678:](https://bugs.earthdata.nasa.gov/browse/CUMULUS-678)

## Changes

* Changes how CMR XML files are handled in move-granules tasks.
* That change allows the move-granules to call into updateCMRMetadata which is a format generic function to update cmr metadata.

## Test Plan
Things that should succeed before merging.

- [ x] Unit tests

### CODE REVIEW

File: tasks/move-granules/index.js [hunk 1]
```diff
@@ -234,60 +244,64 @@ async function moveFileRequest(file, sourceBucket, duplicateHandling, buckets) {
...
+ **/
+async function moveFilesForAllGranules(
+  granulesObject,
+  sourceBucket,
+  duplicateHandling,
+  bucketsConfig
+) {
   const moveFileRequests = Object.keys(granulesObject).map(async (granuleKey) => {
     const granule = granulesObject[granuleKey];
+    // TODO [MHS, 2019-02-05] replace with test for cmrfile
```
Comment (flamingbear) [inline tasks/move-granules/index.js]: This will be done in another PR.

File: tasks/move-granules/index.js [hunk 2]
```diff
@@ -234,60 +244,64 @@ async function moveFileRequest(file, sourceBucket, duplicateHandling, buckets) {
...
     const filesMoved = await Promise.all(
-      filesToMove.map((file) => moveFileRequest(file, sourceBucket, duplicateHandling, buckets))
+      filesToMove.map(
+        (file) => moveFileRequest(file, sourceBucket, duplicateHandling, bucketsConfig)
+      )
+    );
+    const markDuplicates = false;
+    const cmrFilesMoved = await Promise.all(
+      cmrFiles.map(
+        (file) => moveFileRequest(file, sourceBucket, 'replace', bucketsConfig, markDuplicates)
```
Comment (flamingbear) [inline tasks/move-granules/index.js]: This is really the big change here. This is effectively what was happening before, but now it's done first and it's explicit.
a cmr xml file is found in the move-granules task.  We move it from the staging location to it's final location, replacing any file that was there and we don't mark it as a duplicate.

I'm not sure this is the desired behavior, but if it's not, we should add a ticket to change it.

Comment (laurenfrederick) [reply tasks/move-granules/index.js]: I think it's fine to replace without marking as dup. Moving granules via the API will reject the move if a file would be overwritten.

The concern I would have here (and this may still be a concern with the way it's done now) is that the CMR file would get moved and the task could fail before the other files were moved which would have files end up in 2 different locations and you'd lose the CMR file.

Comment (flamingbear) [reply tasks/move-granules/index.js]: I think this is a problem, not sure how you want to address it, but can it be a different ticket?
